src/modules/Elsa.Workflows.Core/Contracts/IWorkflowStateExtractor.cs

```diff
@@ -13,12 +13,12 @@ public interface IWorkflowStateExtractor
     /// <param name="workflowExecutionContext">The workflow execution context to map.</param>
     /// <returns>The mapped workflow state.</returns>
     WorkflowState Extract(WorkflowExecutionContext workflowExecutionContext);
-    
+
     /// <summary>
     /// Applies workflow state to a workflow execution context.
     /// </summary>
     /// <param name="workflowExecutionContext">The workflow execution context to map onto.</param>
     /// <param name="state">The workflow state to apply.</param>
     /// <returns>The updated workflow execution context.</returns>
-    WorkflowExecutionContext Apply(WorkflowExecutionContext workflowExecutionContext, WorkflowState state);
+    Task<WorkflowExecutionContext> Apply(WorkflowExecutionContext workflowExecutionContext, WorkflowState state);
```

Comment:
Please mind the naming convention to apply the `Async` suffix when changing methods into async.